packages/core/src/app.ts

```diff
@@ -322,7 +323,11 @@ export class AppBuilder<TreeNames> {
     let adjustOptions: AdjustImportsOptions = {
       rename,
       extraImports: this.adapter.extraImports(),
-      externalsDir: join(this.root, 'node_modules', '@embroider/externals'),
+
+      // it's important that this is a persistent location, because we fill it
+      // up as a side-effect of babel transpilation, and babel is subject to
+      // persistent caching.
+      externalsDir: join(tmpdir(), 'embroider', 'externals'),
```

Comment:
just heads-up, the OS is allowed to remove files from tmp if they haven't been accessed in some time, if they are missing will we restore them?

Reply:
We will restore them *if* the corresponding babel cache in `$TMPDIR/embroider/webpack-babel-loader` is also missing.

The potential problem is if the babel cache exists and the externals cache does not. In that case, we have no opportunity to rediscover what needs to go in the externals cache because babel will not run again.

Reply:
Entries in the two caches will have very similar access times, so my guess is that this mostly works, and I'm willing to wait for bug reports to find out if it doesn't.

And the cost of trying to do a fancier solution here is pretty high, because it would introduce a new API surface for all stage3 packagers.